src/thunderbots/software/network_input/filter/ball_filter.h

```diff
@@ -4,20 +4,20 @@
 
 #include "ai/world/ball.h"
 #include "geom/point.h"
-#include "util/timestamp.h"
+#include "util/time/timestamp.h"
 
 /**
  * A lightweight datatype used to input new data into the filter.
  * We do this rather than taking the SSL_DetectionBall directly
  * so we can make this module more generic and abstract away
  * the protobuf for testing
  */
-typedef struct
+struct SSLBallDetection
```

Comment:
Why remove the typedef?

Reply:
I didn't see any reason for it, and removing it simplified things a bit (just in terms of syntax, this is semantically the same). Just a little cleanup

Reply:
But doesn't the typedef allow variables to store the struct instance without the `struct` prefix?

https://stackoverflow.com/questions/1675351/typedef-struct-vs-struct-definitions

Reply:
Close, but not the case in C++. If you scroll down a bit in that link you'll see this example:
```
 // C++
struct S { 
    int x; 
}; // S defined as a class

void f( S a ); // correct: struct is optional
```